src/main/java/beam/agentsim/events/handling/BeamEventsWriterXML.java

```diff
@@ -48,22 +54,96 @@ public void init(final String outfilename) {
     public void reset(final int iter) {
     }
 
+    //Modify the method to control logger and based on logging level writing specific attribute
     @Override
     protected void writeEvent(final Event event) {
-        try {
+        String LoggerLevel=this.beamServices.beamConfig().beam().outputs().defaultLoggingLevel();
+
+        if(LoggerLevel.equals(LoggerLevels.OFF.toString())||LoggerLevel.equals("")){System.out.println("No Logs!");}
+        else {
+            try {
             this.out.append("\t<event ");
             Map<String, String> attr = event.getAttributes();
+
             for (Map.Entry<String, String> entry : attr.entrySet()) {
-                this.out.append(entry.getKey());
-                this.out.append("=\"");
-                this.out.append(encodeAttributeValue(entry.getValue()));
-                this.out.append("\" ");
+
+                if(LoggerLevel.equals(LoggerLevels.VERBOSE.toString())){
```

Comment:
The code duplication here is not acceptable.

Reply:
yes , I will rearrange condition.